### DISCUSSION TITLE
Use of subparts

### DISCUSSION THREAD

**Discussion by laurentleb-comnum** [spatie/laravel-permission#2564] [Q&A]
Hi community,

I am implementing roles and permissions to build a two-levels contrl access. I am working on a platform with apps. The first level is about controlling accesses to apps. The second one is about controlling fine access to features inside each app. Could I use subparts pattern in spatie to do that, the main part being the apps level and the subparts being features and sub features inside each app ? If yes, The documentation is really short and it is not clear how to create this hierarchy of roles. Can someone help me ? 
Thanks in advance

https://spatie.be/docs/laravel-permission/v5/basic-usage/wildcard-permissions

**Comment by drbyte** [comment]
Perhaps you can list some examples of your anticipated hierarchy?

It's possible that the Teams feature might be best for the "first level" you mentioned.

**Comment by laurentleb-comnum** [comment]
Thanks for your answer.
For sure, my hierarchy should look like this
User1 can acess to app1, app2. For app1, he has visitor permissions (which means access to restricted features of app1) while on app2 he is administrator (he has access to every features) he can not access to app3

platform
|
|---- app1 OK
|
|--------- feature1 KO
|--------- feature2 KO
|--------- feature3 OK
|--------- feature4 KO
|
|---- app2 OK
|
|--------- feature1 OK
|--------- feature2 OK
|--------- feature3 OK
|--------- feature4 OK
|
|---- app3 KO

Hope it is clear

**Reply by drbyte** [reply]
Yes, I think if you used the Teams feature for each App, making each platform User a member of an app Team, then you could simply set roles/permissions for features within each app/team.

**Reply by laurentleb-comnum** [reply]
Thank you very much, it is an interesting approach. However, I may need teams or organizations management for roles and permissions. I am afraid it conflicts then. 
I saw a real potential in part/subparts but the documentation is too light to use it and be sure.

**Reply by drbyte** [reply]
Can you show the heirarchy you envision for teams/organizations and how that may correlate with apps/features?
